### PR TITLE
Upgrade to Karaf 4.4.5

### PR DESCRIPTION
**Pull request by holgerfriedrich**
For release notes, see:
https://issues.apache.org/jira/secure/ReleaseNote.jspa?version=12353604&projectId=12311140

Refs: https://github.com/openhab/openhab-core/pull/4133

### CODE REVIEW

**Comment by holgerfriedrich** [comment]
@wborn need some time to reorder the runbundles....

**Comment by wborn** [comment]
> need some time to reorder the runbundles

For the itests it doesn't really matter what the bundle order is.

File: pom.xml [hunk 1]
```diff
@@ -74,11 +74,11 @@
     <eea.version>2.2.1</eea.version>
     <jackson.version>2.16.0</jackson.version>
     <jna.version>5.14.0</jna.version>
-    <karaf.version>4.4.4</karaf.version>
+    <karaf.version>4.4.5</karaf.version>
     <netty.version>4.1.104.Final</netty.version>
     <okhttp.version>3.14.9</okhttp.version>
-    <sat.version>0.15.0</sat.version>
-    <slf4j.version>2.0.6</slf4j.version>
+    <sat.version>0.16.0-SNAPSHOT</sat.version>
```
Comment (wborn) [inline pom.xml:80]:
```suggestion
    <sat.version>0.15.0</sat.version>
```

**Comment by wborn** [comment]
Does a local build succeed for you @holgerfriedrich? It does not for me as dynamodb fails:

```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.001 s <<< FAILURE! -- in org.openhab.persistence.dynamodb.internal.TestStoreMixedTypesLegacyTest
[ERROR] org.openhab.persistence.dynamodb.internal.TestStoreMixedTypesLegacyTest -- Time elapsed: 0.001 s <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.eclipse.jetty.server.handler.ResourceHandler$1
	at org.eclipse.jetty.server.handler.ResourceHandler.<init>(ResourceHandler.java:75)
	at com.amazonaws.services.dynamodbv2.local.server.DynamoDBProxyServer.setUpShell(DynamoDBProxyServer.java:58)
	at com.amazonaws.services.dynamodbv2.local.server.DynamoDBProxyServer.<init>(DynamoDBProxyServer.java:41)
	at com.amazonaws.services.dynamodbv2.local.main.ServerRunner.createServer(ServerRunner.java:127)
	at com.amazonaws.services.dynamodbv2.local.main.ServerRunner.createServerFromCommandLineArgs(ServerRunner.java:120)
	at org.openhab.persistence.dynamodb.internal.BaseIntegrationTest.newService(BaseIntegrationTest.java:199)
	at org.openhab.persistence.dynamodb.internal.BaseIntegrationTest.initService(BaseIntegrationTest.java:163)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.NoClassDefFoundError: org/eclipse/jetty/http/compression/HuffmanEncoder [in thread "main"]
	at org.eclipse.jetty.http2.hpack.HpackContext$StaticEntry.<init>(HpackContext.java:469)
	at org.eclipse.jetty.http2.hpack.HpackContext.<clinit>(HpackContext.java:144)
	at org.eclipse.jetty.http2.hpack.HpackFieldPreEncoder.getEncodedField(HpackFieldPreEncoder.java:75)
	at org.eclipse.jetty.http.PreEncodedHttpField.<init>(PreEncodedHttpField.java:103)
	at org.eclipse.jetty.http.PreEncodedHttpField.<init>(PreEncodedHttpField.java:108)
	at org.eclipse.jetty.http2.hpack.HpackEncoder.<clinit>(HpackEncoder.java:83)
	at org.eclipse.jetty.http2.hpack.HpackFieldPreEncoder.getEncodedField(HpackFieldPreEncoder.java:45)
	at org.eclipse.jetty.http.PreEncodedHttpField.<init>(PreEncodedHttpField.java:103)
	at org.eclipse.jetty.http.PreEncodedHttpField.<init>(PreEncodedHttpField.java:108)
	at org.eclipse.jetty.server.ResourceService.<clinit>(ResourceService.java:70)
	... 9 more
```

**Comment by holgerfriedrich** [comment]
@wborn I thought yes, but obviously I must have skipped the tests :-( 
Sorry for that.
I just had to fix jetty version in http binding test, same error here. Not sure how to fix here.... The HuffmannDecoder class was moved around in Jetty....

**Comment by holgerfriedrich** [comment]
@wborn It looks like dynamodb tests are the only remaining issue (mercedesme and hueemulation seem to fail due to restrictive firewall on my build server, I don't see a real problem here).
We have a transitive dependency to jetty 9.4.48, newer packages from 2.x use jetty 11 or 12. 
https://mvnrepository.com/artifact/com.amazonaws/DynamoDBLocal
Jetty team has changed the HuffmanEncoder files.

Do you have any recommendation how to resolve the problem?
- Include the old jetty packages for test
- Try to upgrade to 2.x (needs some changes of dependencies, there is an update script in scripts/fetch_sdk.sh, not sure how to use it)

I have disabled the tests for dynamodb for now, possibly we should merge this way. Currently the add-on build is broken.

**Review by wborn — APPROVED** [review]
Thanks!

**Comment by holgerfriedrich** [comment]
Created follow-up issue to re-enable tests for DynamoDB, #16535.

**Pull request merged**